src/Plugin/Block/ContentLanguageBlock.php

```diff
@@ -86,26 +96,18 @@ public function build() {
       return $build;
     }
 
-    $route_name = $this->pathMatcher->isFrontPage() ? '<front>' : '<current>';
-    $links = $this->languageManager->getLanguageSwitchLinks(LanguageInterface::TYPE_CONTENT, Url::fromRoute($route_name));
-
-    if (isset($links->links)) {
-      // Only show links to the available translation languages except the
-      // current one.
-      $available_languages = array_intersect_key($links->links, $entity->getTranslationLanguages());
-      unset($available_languages[$translation->language()->getId()]);
+    $available_languages = $this->languageProvider->getAvailableEntityLanguages($entity);
 
-      $build = [
-        '#theme' => 'links__oe_multilingual_content_language_block',
-        '#links' => $available_languages,
-        '#attributes' => [
-          'class' => [
-            "language-switcher-{$links->method_id}",
-          ],
+    $build = [
+      '#theme' => 'links__oe_multilingual_content_language_block',
+      '#links' => $available_languages,
+      '#attributes' => [
+        'class' => [
+          "language-switcher",
         ],
-        '#set_active_class' => TRUE,
-      ];
-    }
+      ],
+      '#set_active_class' => TRUE,
+    ];
```

Comment:
Please try this with all the caches enabled. I see no cache metadata here, nor in the `oe_theme_preprocess_links__oe_multilingual_content_language_block` preprocessor. In principle this needs to depend on the current content language context and the cache tags of the entity we get the links for (in case new translations are added to it). No?

Reply:
As discussed, the parent block removes the cache. So a simple comment should suffice.